src/bun.js/webcore/blob.zig

```diff
@@ -1314,7 +1314,39 @@ pub const Blob = struct {
             return .undefined;
         };
 
-        const blob = Blob.findOrCreateFileFromPath(path, globalObject);
+        var blob = Blob.findOrCreateFileFromPath(path, globalObject);
+
+        if (arguments.len == 2) {
+            const opts = arguments[1];
+
+            if (opts.isObject()) {
+                if (opts.getTruthy(globalObject, "type")) |file_type| {
+                    inner: {
+                        if (file_type.isString()) {
+                            var allocator = bun.default_allocator;
+                            var str = file_type.toSlice(globalObject, bun.default_allocator);
+                            defer str.deinit();
+                            var slice = str.slice();
+                            if (!strings.isAllASCII(slice)) {
+                                break :inner;
+                            }
+                            if (vm.mimeType(str.slice())) |entry| {
```

Comment:
if the mimeType exists in the hash table, we avoid cloning the string

https://github.com/oven-sh/bun/blob/a87aa2fafe05c358455173ea65877be5eefcbd63/src/bun.js/webcore/blob.zig#L3011-L3018

I suggest copying from the above pattern instead